public-api/src/main/java/org/pmiops/workbench/cdr/model/QuestionConcept.java

```diff
@@ -0,0 +1,227 @@
+
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+@Entity
+@Table(name = "concept")
+public class QuestionConcept {
+
+    private long conceptId;
+    private String conceptName;
+    private String conceptCode;
+    private String domainId;
+    private long countValue;
+    private float prevalence;
+    private AchillesAnalysis countAnalysis;
+    private AchillesAnalysis genderAnalysis;
+    private AchillesAnalysis ageAnalysis;
+
+    public static final long SURVEY_COUNT_ANALYSIS_ID = 3110;
+    public static final long SURVEY_GENDER_ANALYSIS_ID = 3111;
+    public static final long SURVEY_AGE_ANALYSIS_ID = 3112;
+    public static Map<String, String> ageStratumNameMap  = new HashMap<String, String>();
+    public static Map<String, String> genderStratumNameMap = new HashMap<String, String>();
+
+    /* Take analysis list with results and put them on the list of questions.
+     * This is used when we get a whole list of Analysis with the results for a list of questions from tha dao
+     * so that the return to the ui is a nice question object with analyses and results
+     * Questions updated by reference
+     */
+    public static void mapAnalysesToQuestions(List<QuestionConcept> questions, List<AchillesAnalysis> analyses ) {
+        Map<Long, QuestionConcept> questionMap = new HashMap<Long, QuestionConcept>();
+        for (QuestionConcept q : questions) {
+            questionMap.put(q.getConceptId(), q);
+        }
+        setAgeStratumNameMap();
+        setGenderStratumNameMap();
+        for (AchillesAnalysis analysis : analyses) {
+            // Add stratum5Name to the results for the ui -- ie Male, Female , Age Decile name
+            for (AchillesResult r : analysis.getResults()) {
+                // Add analysis to question if need to
+                Long qid = Long.valueOf(r.getStratum2());
+                QuestionConcept q = questionMap.get(qid);
+
+                if ( q.getAnalysis(analysis.getAnalysisId())  == null) {
+                    q.setAnalysis(new AchillesAnalysis(analysis));
+                }
+                AchillesAnalysis questionAnalysis = q.getAnalysis(analysis.getAnalysisId());
+                questionAnalysis.addResult(r);
+
+                if (r.getStratum5Name() == null || r.getStratum5Name() == "") {
+                    if (analysis.getAnalysisId() == SURVEY_AGE_ANALYSIS_ID) {
+                        r.setStratum5Name(ageStratumNameMap.get(r.getStratum5()));
+                    }
+                    if (analysis.getAnalysisId() == SURVEY_GENDER_ANALYSIS_ID) {
+                        r.setStratum5Name(genderStratumNameMap.get(r.getStratum5()));
+                    }
+                }
+            }
+        }
+    }
+
+    /* Todo Find right place for these static things to be generated from db if possible and live */
+    public static void setAgeStratumNameMap() {
```

Comment:
You can call these in a static initializer if you prefer:

static {
  setAgeStratumNameMap();
  setGenderStratumNameMap();
}

and then not bother calling them in mapAnalysesToQuestions. That way this only runs once.

Reply:
Thanks. Fixed. I'm testing now and will push.